src/profiles/ro_crate/ro_crate_parser.py

```diff
@@ -66,15 +90,39 @@ def __init__(
         with open(CRATE_TO_TARDIS_PROFILE, encoding="utf-8") as f:
             self.mapper: CrateToTardisMapper = CrateToTardisMapper(json.load(f))
         self.crate = ROCrate(Path(crate_root_path))
-        self.crate_name = crate_name
-        if not self.crate_name:
-            # To be replaced with UUID defined in RO-Crate root
-            self.crate_name = self.crate.source.parts[-2]
-
+        self.name = crate_name
+        self.name = self._read_crate_name()
```

Comment:
I see now that `self._read_crate_name()` is checking the member. It's a bit confusing that in the case where `crate_name` has been supplied, we assign it twice in different places. I wonder if it would be more readable to have:
```python
self.name = crate_name or self._read_crate_name()
```
and then remove this line from `_read_crate_name`:
```python
if self.name:
    return self.name
```

Reply:
yeah that's a lot cleaner